src/NServiceBus.Core/Persistence/PersistenceExtentions.cs

```diff
@@ -19,10 +19,21 @@ public class PersistenceExtentions<T, S> : PersistenceExtentions<T>
         /// <summary>
         /// </summary>
         /// <param name="settings"></param>
-        public PersistenceExtentions(SettingsHolder settings)
-            : base(settings)
+        public PersistenceExtentions(SettingsHolder settings) : base(settings, typeof(S))
         {
         }
+
+        /// <summary>
+        /// Defines the list of specific storage needs this persistence should provide
+        /// </summary>
+        /// <param name="specificStorages"></param>
+        /// <returns></returns>
+        /// <exception cref="InvalidOperationException"></exception>
+        [ObsoleteEx(RemoveInVersion = "7.0", TreatAsErrorFromVersion = "5.2")]
+        public new PersistenceExtentions<T> For(params Storage[] specificStorages)
+        {
+            throw new InvalidOperationException("Do not invoke .For() when StorageType is already specified.");
```

Comment:
Wasn't sure what would be the best way to prevent the following scenario https://github.com/Particular/NServiceBus/issues/2622#issuecomment-69142522, so decided to throw.